src/task.cpp

```diff
@@ -142,6 +142,34 @@ task::get_environ(size_t max_size) const
     return environ;
 }
 
+io_stat task::get_io() const {
+    static const std::string STAT_FILE("io");
+    auto path = _task_root + STAT_FILE;
+    std::ifstream io(path);
```

Comment:
The entire code bases reuses a set of parsers that handle most of the existing cases in the procfs.
For example, to parse this file, that looks like:
```
rchar: 1948
wchar: 0
syscr: 7
...
```
You could've used something similar to the `proc_status.cpp` parser.
Check out `status_parser` @ `parsers.hpp` and the implementation, it should be really easy to reuse for the `io` file as well.
The parser accepts a dictionary of keys with a simple function pointer for every key to parse the line.

There are two important advantages to using that parser in comparison to your current method (which is elegant and very cpp-y):
1. Retains consistency with the current code. There's a single way to do things, making it easy to expand and maintain.
2. If the order of the keys in the `io` file ever changes, this code will fail miserably, whereas the `status_parser`-like will work out of the box. Not assuming anything about the order is very important for parsers, especially when keys might be added, moved or removed.

Reply:
Thanks for these comments, I updated the pull request :-)